modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -112,13 +117,26 @@ class Instrument_Manager extends \NDB_Menu_Filter
      */
     protected function handlePOST(ServerRequestInterface $request): ResponseInterface
     {
-        if (!$this->canWriteFiles()) {
+        // Ensure the user is allowed to upload.
+        if (! \User::singleton()->hasPermission('instrument_manager_write')) {
             return (new \Zend\Diactoros\Response())
                 ->withStatus(403, 'Forbidden')
                 ->withBody(
                     new \LORIS\Http\StringStream(
                         json_encode(
-                            array('error' => CANNOT_WRITE_FILES)
+                            array('error' => 'Forbidden')
+                        )
+                    )
+                );
+        }
+        // Ensure the server is properly configured.
+        if (! $this->canWriteFiles()) {
+            return (new \Zend\Diactoros\Response())
```

Comment:
I think this should be an http\response not Zend.
Would be good to change it in line 122 at the same time